src/cli_shared/cli/mod.rs

```diff
@@ -162,13 +162,21 @@ impl CliOpts {
             None => Config::default(),
         };
 
-        if let Some(chain) = &self.chain {
-            // override the chain configuration
-            cfg.chain = Arc::new(ChainConfig::from_chain(chain));
-        } else {
-            // override any custom changes to the chain configuration based on the used
-            // network.
-            cfg.chain = Arc::new(ChainConfig::from_chain(&cfg.chain.network));
+        // override chain specific configurations
+        let chain = self.chain.clone().unwrap_or(cfg.chain.network.clone());
```

Comment:
Maybe `unwrap_or_else` like in your other PR. :)

Reply:
Fixed